### PR TITLE
Entity-Component-System interface

### PR DESCRIPTION
**Pull request by jaafersheriff**
It works

### CODE REVIEW

**Comment by barlebino** [comment]
I think it's good

**Comment by alacardnightmare** [comment]
More Comments. JK.

File: src/Engine/Scene/Scene.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        enum SystemType{
+            GAMELOGIC
+        };
+        Scene();
+
+        /* Game Objects */
+        GameObject* createGameObject();
+        void addGameObject(GameObject *);
+    
+        /* Components */
```
Comment (Daskie) [inline src/Engine/Scene/Scene.hpp]: Learn how to use templates noob

**Comment by alacardnightmare** [comment]
A good start. Keep up the good work, chap. Approved.

**Pull request merged**